fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexValidator.java

```diff
@@ -44,5 +46,49 @@ public void validate(@Nonnull MetaDataValidator metaDataValidator) {
         for (RecordType recordType : metaDataValidator.getRecordMetaData().recordTypesForIndex(index)) {
             LuceneIndexExpressions.validate(index.getRootExpression(), recordType.getDescriptor());
         }
+
+        validateIndexOptions(index);
+    }
+
+    @VisibleForTesting
+    public static void validateIndexOptions(@Nonnull Index index) {
+        validateAnalyzerNamePerFieldOption(LuceneIndexOptions.LUCENE_ANALYZER_NAME_PER_FIELD_OPTION, index);
+        validateAnalyzerNamePerFieldOption(LuceneIndexOptions.AUTO_COMPLETE_ANALYZER_NAME_PER_FIELD_OPTION, index);
+        validateAutoCompleteExcludedFields(index);
+    }
+
+    private static void validateAnalyzerNamePerFieldOption(@Nonnull String optionKey, @Nonnull Index index) {
+        String analyzerNamePerFieldOption = index.getOption(optionKey);
+        if (analyzerNamePerFieldOption != null) {
+            analyzerNamePerFieldOption = analyzerNamePerFieldOption.strip();
```

Comment:
I don't think the way that `strip()` is handled here is quite right, because `strip` is not called when the index options are actually evaluated later

Reply:
Changed it to also call the ```strip()``` when actually evaluating the index options.